readme.md

```diff
@@ -0,0 +1,77 @@
+## wasm-cli-plugin
+
+This is a Docker CLI plugin that allows running `wasi/wasm` applications built from Dockerfiles and distributed with Docker registries on any platform as an unprivileged user.
+
+### Getting started
+
+Running `make shell` puts you in a development environment where you have access to `docker`, `buildx` and the example apps in this repository.
+
+#### Available commands
+
+The UI is similar to docker but simplified a lot. You can pull an image with `docker wasm pull <image>` and run with `docker wasm run <image> <cmd>`. 
+
+```
+# docker wasm
+
+Usage:	docker wasm [OPTIONS] COMMAND
+
+Run wasm containers
+
+Options:
+      --data-root string   Root directory of persistent state (default "/root/.docker/wasm")
+  -D, --debug              Enable debug logs
+
+Commands:
+  ls          List wasm images
+  pull        Pull a wasm image
+  rm          Remove a wasm image
+  run         Run a wasm image
+```
+
+### Building
+
+```
+make binaries
+```
+
+Builds `docker-wasm` binary under `./bin` folder. 
+
+### Installing
+
+Build `docker-wasm` or pull from [releases](https://github.com/tonistiigi/wasm-cli-plugin/releases) place it on `PATH`. Either [`wasmtime`](https://github.com/CraneStation/wasmtime) or [`wasmer`](https://github.com/wasmerio/wasmer/releases) runtime needs to be on `PATH` to run wasi applications. On Linux and OSX `wasmtime` runtime binary is also built with `docker-wasm`. 
+
+`docker-wasm` binary can be used directly or since `v19.03` it can be used as a Docker CLI plugin, adding a `docker wasm` command to Docker CLI. To install the plugin copy the built binary to `~/.docker/cli-plugins/docker-wasm`.
+
+
+### Building multi-platform images with wasm support
+
+Wasm applications are built from Dockerfiles by setting `--platform=wasi/wasm` flag on build time. `--platform` is available (without experimental) since Docker 19.03 (with BuildKit). Docker [buildx]() plugin is another way to build and allows specifying multiple platforms together into a single multi-platform image and works with previous Docker engine versions.
```

Comment:
```suggestion
Wasm applications are built from Dockerfiles by setting `--platform=wasi/wasm` flag on build. `--platform` is available (without experimental) since Docker 19.03 (only with BuildKit). Docker [buildx](https://github.com/docker/buildx) plugin is another way to build and allows specifying multiple platforms together into a single multi-platform image and works with previous Docker engine versions.
```